test/unit/handlebars.cpp

```diff
@@ -0,0 +1,460 @@
+//
+// Copyright (c) 2023 alandefreitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0.
+// https://www.boost.org/LICENSE_1_0.txt
+//
+
+#include <mrdox/Support/Handlebars.hpp>
+#include <mrdox/Support/Path.hpp>
+#include <mrdox/Support/String.hpp>
+#include <mrdox/Support/Dom.hpp>
+#include "test_macros.hpp"
+#include "diff.hpp"
+#include <fmt/format.h>
+#include <ranges>
+
+void splitLines(std::string_view text, std::vector<std::string_view> &lines);
+
+using namespace clang::mrdox;
+
+template<>
+class fmt::formatter<dom::Kind, char> {
+public:
+    template<typename ParseContext>
+    constexpr auto parse(ParseContext& ctx) {
+        return ctx.begin();
+    }
+
+    template<typename FormatContext>
+    auto format(dom::Kind const& value, FormatContext& ctx) {
+        switch (value) {
+            case dom::Kind::Null:
+                return format_to(ctx.out(), "null");
+            case dom::Kind::Boolean:
+                return format_to(ctx.out(), "boolean");
+            case dom::Kind::Integer:
+                return format_to(ctx.out(), "integer");
+            case dom::Kind::String:
+                return format_to(ctx.out(), "string");
+            case dom::Kind::Array:
+                return format_to(ctx.out(), "array");
+            case dom::Kind::Object:
+                return format_to(ctx.out(), "object");
+        }
+        return format_to(ctx.out(), "unknown");
+    }
+};
+
+
+int
+main() {
+    /////////////////////////////////////////////////////////////////
+    // Fixtures
+    /////////////////////////////////////////////////////////////////
+    std::string_view template_path = MRDOX_UNIT_TEST_DIR "/fixtures/handlebars_features_test.adoc.hbs";
+    std::string_view partial_paths[] = {
+        MRDOX_UNIT_TEST_DIR "/fixtures/record-detail.adoc.hbs",
+        MRDOX_UNIT_TEST_DIR "/fixtures/escaped.adoc.hbs"};
+    std::string_view output_path = MRDOX_UNIT_TEST_DIR "/fixtures/handlebars_features_test.adoc";
+    std::string_view error_output_path = MRDOX_UNIT_TEST_DIR "/fixtures/handlebars_features_test_error.adoc";
+    auto template_text_r = files::getFileText(template_path);
+    REQUIRE(template_text_r);
+    auto master_file_contents_r = files::getFileText(output_path);
+    auto template_str = *template_text_r;
+    REQUIRE_FALSE(template_str.empty());
+
+    HandlebarsOptions options;
+    options.noHTMLEscape = true;
+
+    /////////////////////////////////////////////////////////////////
+    // Context
+    /////////////////////////////////////////////////////////////////
+    dom::Object context;
+    dom::Object page;
+    page.set("kind", "record");
```

Comment:
Oh wow this whole test looks like a hot mess to maintain... why can't we do something like what we did for mrdox where you have some input data, a small template, and an expected output file then the test runs handlebars on the template and compares the actual output from the expected output file. Then you never need to change the C++ to add tests.

Reply:
That's kind of what I would do if the project was more independent. I would have libraries to parse json, some nice unit testing framework, etc... I'm doing things this way to be less intrusive. I can't just add things to cmakelists.txt, etc.